src/include/cache_inline.h

```diff
@@ -445,6 +445,22 @@ __wt_cache_full(WT_SESSION_IMPL *session)
     return (__wt_cache_bytes_inuse(cache) >= conn->cache_size);
 }
 
+/*
+ * __wt_cache_almost_full --
+ *     Return if the cache is almost reaching capacity.
+ */
+static inline bool
+__wt_cache_almost_full(WT_SESSION_IMPL *session)
+{
+    WT_CACHE *cache;
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+    cache = conn->cache;
+
+    return (__wt_cache_bytes_inuse(cache) >= ((conn->cache_size / 100) * 90));
```

Comment:
The "90" here should be replaced by the `eviction_trigger` value - it also seems as though a cache flag is probably already tracking it (WT_CACHE_EVICT_CLEAN_HARD and WT_CACHE_EVICT_DIRTY_HARD).

As I mentioned earlier - I think this change needs to be more aware of the distinction between clean and dirty data. It should probably focus on the dirty content - since that's where we are seeing problems (reconciling a page from a table is reading in and dirtying a lot of history store content).

Reply:
Point noted. I'll look into either using existing cache flags or including dirty ratio in calculations here (or both).